Tensile/Source/client/main.cpp

```diff
@@ -83,124 +83,524 @@ namespace Tensile
+                                                                                                "Op"
+                                                                                                "er"
+                                                                                                "at"
+                                                                                                "io"
+                                                                                                "ns"
+                                                                                                " a"
+                                                                                                "pp"
+                                                                                                "li"
+                                                                                                "ed"
+                                                                                                " t"
+                                                                                                "o "
+                                                                                                "D"
+                                                                                                ".")
+
+                                                                                                ("p"
+                                                                                                 "r"
+                                                                                                 "o"
+                                                                                                 "b"
+                                                                                                 "l"
+                                                                                                 "e"
+                                                                                                 "m"
+                                                                                                 "-"
+                                                                                                 "s"
+                                                                                                 "t"
+                                                                                                 "a"
+                                                                                                 "r"
+                                                                                                 "t"
+                                                                                                 "-"
+                                                                                                 "i"
+                                                                                                 "d"
```

Comment:
The formatting rules need to be tweaked, some of the files are coming out pretty messy like this one. Another spot is comments in many of the testing files which list parameter sets being treated as code and becoming harder to read.

Reply:
Not everything can be handled by clang format.  Turning off may be best for select sections.

// clang-format off
...
// clang-format on

Reply:
> 
> 
> The formatting rules need to be tweaked, some of the files are coming out pretty messy like this one. Another spot is comments in many of the testing files which list parameter sets being treated as code and becoming harder to read.
```
// clang-format off 
...
// clang-format on
```
... is sometimes necessary.

Reply:
What about the Doxygen comments? They are scattered throughout the codebase and turning formatting off and then on for each one will be a lot of work.

Reply:
> What about the Doxygen comments? They are scattered throughout the codebase and turning formatting off and then on for each one will be a lot of work.

Does `clang-format` not already recognize Doxygen comments?

We use Doxygen extensively in rocBLAS, and have never seen any problem.

The `clang-format` settings we have used in rocBLAS (copied to Tensile), make the user's formatting of comments have first priority. Meaning, unless a comment line is too long and must be wrapped, it will be left alone. Doxygen comments less than a certain line length will therefore be left alone.